wsl-vpnkit-setup.sh

```diff
@@ -33,28 +36,31 @@ if [ -z "${WSL_DISTRO_NAME:+set}" ]; then
   WSL_DISTRO_NAME="$(IFS='\'; x=($(wslpath -w /)); echo "${x[${#x[@]}-1]}")"
 fi
 
-# Determine dependencies
-dependencies=(socat)
-deb_install=(socat)
-if [ "${no_docker}" = "0" ]; then
-  dependencies+=(unzip isoinfo)
-  deb_install+=(unzip genisoimage)
-fi
-
-for cmd in "${dependencies[@]}"; do
-  if ! command -v "${cmd}" &> /dev/null; then
-    if command -v apt &> /dev/null; then
-      apt update
-      apt install -y "${deb_install[@]}"
-      break
-    # elif command -v yast2 &> /dev/null; then
-    #   ...
-    else
-      echo "Todo: program other package managers" &> /dev/null
+function install_socat()
+{
+  if command -v apt > /dev/null 2>&1; then
+    apt update
+    apt install -y socat
+  else
+    echo "There is no automated solution to install \"socat\" on this OS" >&2
+    read -pr "Please enter a command to install \"socat\": " cmd
+    eval "${cmd}"
+    if ! command -v socat; then
+      echo "socat does not appear to be installed. Please get socat installed and try again, or try using --on-vpn"
       exit 3
     fi
   fi
-done
+}
+
+if ! command -v socat &> /dev/null; then
+  if [ "${on_vpn}" = "0" ]; then
+    install_socat
+  else
+    # This appears to work in alpine (musl) and ubuntu/fedora alike (glibc)
+    download_ps https://github.com/andrew-d/static-binaries/raw/8ae38c79510d072cdba0bf719ef4f16c052e2abc/binaries/linux/x86_64/socat /usr/local/bin/socat
+    chmod 755 /usr/local/bin/socat
+  fi
+fi
```

Comment:
This is pretty cool. and removes the dependency on connecting WSL to the internet. Do you think this would work on Ubuntu?

Reply:
I'll try to test this on a few ubuntu version and update on the weekend if it works

Reply:
👍 Yes. This static binary will work on Ubuntu and Debian and even Alpine and Fedora or RHEL etc... What limits us to "Just Debian and Ubuntu" is the .service file. I dug a little and found out that this init system (called openrc) varies too much from OS to OS (the support scripts/commands aren't all the same) and so if we just say "this works on Debian and Ubuntu" we'll be good. (As this is only a temporary fix, we can wait to see if someone has a real reason they "can't" install Debian/Ubuntu along side whatever distro they want to use, and try to assist accordingly.)

Let me know how the test goes!

Reply:
Apologies for the late reply, been a hectic few weeks. I have only managed to test this on two version of Ubuntu + Kali. Seems to be working fine.